doc/_scripts/software_maturity_scanner.py

```diff
@@ -0,0 +1,554 @@
+
+    # Find all SoCs from integration_platforms in sample.yaml
+    all_socs = find_all_socs(all_samples)
+
+    # Check which symbols are experimental by looking through build.log files
+    build_logs = list(args.build_output.glob("**/build.log"))
+    exp_soc_sets = find_experimental_symbols(build_logs)
+
+    # Each feature is associated with a dictionary mapping each supported SoC
+    # to its state (Experimental/Supported)
+    soc_sets = {}
+    top_table_info = input_doc["top_table"]
+    for tech in top_table_info:
+        soc_sets[tech] = set()
+
+    for feature in all_features:
+        soc_sets[feature] = {}
+
+    # Look for the desired symbol combinations in all .config files
+    all_conf_files = args.build_output.glob("**/zephyr/.config")
+    for conf_file in all_conf_files:
+        kconf = kconfiglib.Kconfig(filename=conf_file)
+        soc = kconf.variables["CONFIG_SOC"].value
+        soc = soc.strip('"').split("_")[0].lower()
+
+        # Look for technology symbols
+        for tech, rule in top_table_info.items():
+            rule = parse_rule(rule)
+            # Top level technologies are currently only Supported/Not supported
+            if evaluate_rule(kconf.variables, rule):
+                soc_sets[tech].add(soc)
+
+        # Look for feature symbols
+        for feature, rule in all_features.items():
+            if evaluate_rule(kconf.variables, rule):
+                experimental_symbols = exp_soc_sets[soc] if soc in exp_soc_sets else set()
+                status = find_maturity_level(rule, experimental_symbols)
+                soc_sets[feature][soc] = status
+
+    # Create the output data structure
+    top_table = {}
+    for tech in sorted(top_table_info):
+        top_table[tech] = {"Supported": [], "Experimental": []}
+        for soc in all_socs:
+            if soc in soc_sets[tech]:
+                top_table[tech]["Supported"].append(soc)
+
+    # Feature tables
+    category_results = {}
+    for cat in sorted(feature_categories):
+        features = feature_categories[cat]
+        table = {}
+
+        for feature_name in sorted(features):
+            feature = f"{cat}_{feature_name}"
+            table[feature] = {"Supported": [], "Experimental": []}
+
+            for soc in all_socs:
+                if soc in soc_sets[feature]:
+                    status = soc_sets[feature][soc]
+                    table[feature][status].append(soc)
+
+        category_results[cat] = table
+
+    output = {
+        "all_socs": all_socs,
+        "top_level": top_table,
+        "features": category_results,
+    }
+
+    # Upload to Azure
+    if args.publish_results:
+        cc = ContainerClient.from_connection_string(
+            AZ_CONN_STR.format(args.access_key), AZ_CONTAINER
+        )
+
+        filename = UPLOAD_FILENAME.format(args.commit_sha)
+        upload_data(filename, output, cc)
+
+    # Save to local file
+    if args.save_output:
```

Comment:
please save_output first, and then try publishing results (easier to debug if publishing fails)